apps/core-app/src/pages/Proposals.tsx

```diff
@@ -1,43 +1,130 @@
+import { ChangeEvent, MouseEvent, useEffect, useMemo, useState } from 'react';
 import styled from 'styled-components';
 import {
-  BiColumnLayout,
   Button,
-  Card,
-  widthQuery,
   Dialog,
   DialogTrigger,
   DialogContent,
+  SingleColumnLayout,
 } from '@daohaus/ui';
+import { statusFilter } from '@daohaus/dao-data';
 import { BsPlusLg } from 'react-icons/bs';
 
-import { useProposals } from '../contexts/DaoContext';
+import { defaultDaoData, useDao, useProposals } from '../contexts/DaoContext';
 import { NewProposalList } from '../components/NewProposalList';
 import { FORM } from '../legos/form';
-import { useMemo } from 'react';
+import SearchInput from '../components/SearchInput';
+import FilterDropdown from '../components/FilterDropdown';
+import { BaseProposalCard } from '../components/BaseProposalCard';
+import useDebounce from '../utils/debounceHook';
+import { PROPOSAL_STATUS } from '@daohaus/common-utilities';
 
-const LeftCard = styled(Card)`
+const ActionsContainer = styled.div`
   width: 100%;
-  min-width: 54rem;
-  max-width: 64rem;
-  @media ${widthQuery.md} {
-    max-width: 100%;
-    min-width: 0;
-  }
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 3rem;
+`;
+
+const SearchFilterContainer = styled.div`
+  display: flex;
+  gap: 2.1rem;
 `;
 
 export const VALID_NEW_PROPOSALS = [FORM.SIGNAL, FORM.SHARE_SWAP];
 
 export function Proposals() {
-  const { proposals } = useProposals();
+  const {
+    proposals,
+    setProposalsPaging,
+    proposalsNextPaging,
+    setProposalsFilter,
+    setProposals,
+  } = useProposals();
+  const { dao } = useDao();
+  const [searchTerm, setSearchTerm] = useState<string | ''>('');
+  const [filter, setFilter] = useState<string | ''>('');
+
+  const debouncedSearchTerm = useDebounce<string>(searchTerm, 700);
 
   const newProposals = useMemo(() => {
+    console.log('proposals', proposals);
     return Object.keys(FORM).map((key) => FORM[key]);
-  }, []);
+  }, [proposals]);
+
+  useEffect(() => {
+    console.log('deboun', debouncedSearchTerm);
+
+    const filterQuery =
+      filter !== ''
+        ? statusFilter(
+            PROPOSAL_STATUS[filter],
+            Number(dao?.votingPeriod) + Number(dao?.gracePeriod)
+          )
+        : undefined;
+
+    if (debouncedSearchTerm && debouncedSearchTerm !== '') {
```

Comment:
I believe those checks are the same. An empty string is always false as a boolean

Maybe check against the length of the string? If you decide to have the initial state as null you would want to check against that also.
```suggestion
    if (debouncedSearchTerm && debouncedSearchTerm.length > 0) {
```